agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -804,6 +816,30 @@ bool main_thread::backhaul_fsm_main(bool &skip_select)
     return (true);
 }
 
+bool main_thread::send_1905_topology_discovery_message()
+{
+    auto cmdu_hdr = cmdu_tx.create(0, ieee1905_1::eMessageType::TOPOLOGY_DISCOVERY_MESSAGE);
+    if (!cmdu_hdr) {
+        LOG(ERROR) << "Failed to create TOPOLOGY_DISCOVERY_MESSAGE cmdu";
+        return false;
+    }
+    auto tlvAlMac = cmdu_tx.addClass<ieee1905_1::tlvAlMacAddressType>();
+    if (!tlvAlMac) {
+        LOG(ERROR) << "Failed to create tlvAlMacAddressType tlv";
+        return false;
+    }
+    tlvAlMac->mac() = network_utils::mac_from_string(bridge_info.mac);
+    auto tlvMac     = cmdu_tx.addClass<ieee1905_1::tlvMacAddress>();
+    if (!tlvMac) {
+        LOG(ERROR) << "Failed to create tlvMacAddress tlv";
+        return false;
+    }
+    tlvMac->mac() = network_utils::mac_from_string(bridge_info.mac);
```

Comment:
Meh. This is wrong. The MAC should be the MAC address of the interface on which you send the message: "MAC address of the interface on which the message is transmitted".

Also it's a neighbour multicast message, which I don't think the transport supports at the moment.

Oh well, minimal implementation I guess...